README.md

```diff
@@ -43,19 +43,38 @@ The basic sequence for driving the Bulk API is:
 
 ## Bulk Query
 
-`SalesforceBulk.create_query_job(object_name, contentType='CSV', concurrency=None)`
+`bulk.create_query_job(object_name, contentType='CSV')`
 
 Example
 
 ```
+import unicodecsv
 job = bulk.create_query_job("Contact", contentType='CSV')
 batch = bulk.query(job, "select Id,LastName from Contact")
-while not bulk.is_batch_done(job, batch):
+bulk.close_job(job)
+while not bulk.is_batch_done(batch):
 	sleep(10)
+
+for result in bulk.get_all_results_for_query_batch(batch):
+	with unicodecsv.DictReader(result, encoding='utf-8') as reader:
+        for row in reader:
+            print row # dictionary rows
+```
+
+Same example but for JSON
+```
+from salesforce_bulk.util import IteratorBytesIO
+import json
+job = bulk.create_query_job("Contact", contentType='JSON')
+batch = bulk.query(job, "select Id,LastName from Contact")
 bulk.close_job(job)
+while not bulk.is_batch_done(batch):
+	sleep(10)
 
-for row in bulk.get_batch_result_iter(job, batch, parse_csv=True):
-	print row   #row is a dict
+for result in bulk.get_all_results_for_query_batch(batch):
+    result = json.load(IteratorBytesIO(result))
+    for row in result:
+        print row # dictionary rows
 ```
```

Comment:
It would be good to mention that JSON is the better format due to its proper null handling. Better to put the JSON example before the CSV example.